### PR TITLE
Add checks for tunes links if they are used

### PR DESCRIPTION
**Pull request by meyskens**
This route is made for a cron-based script that will ask ITFrame if it uses a specific link so it can remove old and unused ones.

---
This change is [Reviewable](https://reviewable.io/reviews/innovate-technologies/itframe/17)

### CODE REVIEW

File: components/tunes/personalMusicDatabase.js [hunk 1]
```diff
@@ -207,46 +201,21 @@ module.exports.calculateUsedSpace = (username, callback) => {
...
+module.exports.isLinkInUse = async (link) => {
+    const internalURLs = await TunesPersonalModel.find({ internalURL: link })
+    if (internalURLs.length !== 0) {
+        return true
     }
+    const findArray = []
+    for (let bitrate of [32, 64, 98, 128, 192, 265, 320]) {
+        const find = {}
+        find[`processedURLS.${bitrate}`] = link
+        findArray.push(find)
```
Comment (meyskens) [inline components/tunes/personalMusicDatabase.js]: ESLint didn't like that.

**Comment by leoetlino** [comment]
Reviewed 2 of 2 files at r1.
Review status: 1 of 2 files reviewed at latest revision, 7 unresolved discussions.

---

*[components/tunes/personalMusicDatabase.js, line 215 \[r1\]](https://reviewable.io:443/reviews/innovate-technologies/itframe/17#-KNY_PWeliKf3SEIBr9K:-KNY_PWeliKf3SEIBr9L:929649161) ([raw file](https://github.com/innovate-technologies/itframe/blob/e77aa355a068bf395e02de2fc538382502c8d9d8/components/tunes/personalMusicDatabase.js#L215)):*
> ```JavaScript
>         findArray.push(find)
>     }
>     const processedURLS = await TunesPersonalModel.find({$or: findArray})
> ```

`processedUrls`

---

*[components/tunes/personalMusicDatabase.js, line 219 \[r1\]](https://reviewable.io:443/reviews/innovate-technologies/itframe/17#-KNY_Sq7mizFHc7JbjCY:-KNY_Sq7mizFHc7JbjCZ:-2049957800) ([raw file](https://github.com/innovate-technologies/itframe/blob/e77aa355a068bf395e02de2fc538382502c8d9d8/components/tunes/personalMusicDatabase.js#L219)):*
> ```JavaScript
>         return true
>     }
>     return false
> ```

`return processedUrls.length !== 0`

---

*[http/internal/tunes.js, line 142 \[r1\]](https://reviewable.io:443/reviews/innovate-technologies/itframe/17#-KNY_ebdl4ylwG_fdo3p:-KNY_ebdl4ylwG_fdo3q:-1630119237) ([raw file](https://github.com/innovate-technologies/itframe/blob/e77aa355a068bf395e02de2fc538382502c8d9d8/http/internal/tunes.js#L142)):*
> ```JavaScript
>     });
> 
>     app.get("/tunes/is-link-in-use/" + config.tunesKey, wrap( async (req, res, next) => {
> ```

no space before async: `wrap(async`…

---

*[http/internal/tunes.js, line 144 \[r1\]](https://reviewable.io:443/reviews/innovate-technologies/itframe/17#-KNY_lbE3wbfE7oEbqpQ:-KNY_lbE3wbfE7oEbqpR:1861010866) ([raw file](https://github.com/innovate-technologies/itframe/blob/e77aa355a068bf395e02de2fc538382502c8d9d8/http/internal/tunes.js#L144)):*
> ```JavaScript
>     app.get("/tunes/is-link-in-use/" + config.tunesKey, wrap( async (req, res, next) => {
>         if (!req.query.link) {
>             return next(new Error("No link provided"));
> ```

there's no need for next for error handling in an async route handler. Just `throw` it

---

*[http/internal/tunes.js, line 148 \[r1\]](https://reviewable.io:443/reviews/innovate-technologies/itframe/17#-KNY_sI1y7HVXIAGPAaX:-KNY_sI1y7HVXIAGPAaY:322948653) ([raw file](https://github.com/innovate-technologies/itframe/blob/e77aa355a068bf395e02de2fc538382502c8d9d8/http/internal/tunes.js#L148)):*
> ```JavaScript
>         try {
>             const isInUse = await db.isLinkInUse(req.query.link)
>             res.json({isInUse})
> ```

space after { and before }

---

*[http/internal/tunes.js, line 151 \[r1\]](https://reviewable.io:443/reviews/innovate-technologies/itframe/17#-KNY_yydKCRX0DiatCsL:-KNY_yydKCRX0DiatCsM:-631080163) ([raw file](https://github.com/innovate-technologies/itframe/blob/e77aa355a068bf395e02de2fc538382502c8d9d8/http/internal/tunes.js#L151)):*
> ```JavaScript
> 
>         } catch (error) {
>             return next(error)
> ```

actually, you don't need to try and catch if you aren't going to change the error. just let the wrapper function catch it and pass it to the error handler.

Since this isn't the case here, you can get rid of the whole try/catch block.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/innovate-technologies/itframe/17)*

**Comment by meyskens** [comment]
Review status: 0 of 2 files reviewed at latest revision, 7 unresolved discussions.

---

*[components/tunes/personalMusicDatabase.js, line 213 \[r1\]](https://reviewable.io:443/reviews/innovate-technologies/itframe/17#-KNYPKG91_7J0zEHhUqK-r1-213:-KNYbKqA3_iJs-KQNKrq:-499146009) ([raw file](https://github.com/innovate-technologies/itframe/blob/e77aa355a068bf395e02de2fc538382502c8d9d8/components/tunes/personalMusicDatabase.js#L213)):*
<details><summary><i>Previously, meyskens (Maartje Eyskens) wrote…</i></summary>
> ESLint didn't like that. 
</details>
Done.

---

*[components/tunes/personalMusicDatabase.js, line 215 \[r1\]](https://reviewable.io:443/reviews/innovate-technologies/itframe/17#-KNY_PWeliKf3SEIBr9K:-KNYbLbveGt1HPWeV6PS:-499146009) ([raw file](https://github.com/innovate-technologies/itframe/blob/e77aa355a068bf395e02de2fc538382502c8d9d8/components/tunes/personalMusicDatabase.js#L215)):*
<details><summary><i>Previously, leoetlino (Léo Lam) wrote…</i></summary>
> `processedUrls`
</details>
Done.

---

*[components/tunes/personalMusicDatabase.js, line 219 \[r1\]](https://reviewable.io:443/reviews/innovate-technologies/itframe/17#-KNY_Sq7mizFHc7JbjCY:-KNYbMCg-jM64wz1Fm6Z:-499146009) ([raw file](https://github.com/innovate-technologies/itframe/blob/e77aa355a068bf395e02de2fc538382502c8d9d8/components/tunes/personalMusicDatabase.js#L219)):*
<details><summary><i>Previously, leoetlino (Léo Lam) wrote…</i></summary>
> `return processedUrls.length !== 0`
</details>
Done.

---

*[http/internal/tunes.js, line 142 \[r1\]](https://reviewable.io:443/reviews/innovate-technologies/itframe/17#-KNY_ebdl4ylwG_fdo3p:-KNYbN5zmlcj0CxMX1CD:-499146009) ([raw file](https://github.com/innovate-technologies/itframe/blob/e77aa355a068bf395e02de2fc538382502c8d9d8/http/internal/tunes.js#L142)):*
<details><summary><i>Previously, leoetlino (Léo Lam) wrote…</i></summary>
> no space before async: `wrap(async`…
</details>
Done.

---

*[http/internal/tunes.js, line 144 \[r1\]](https://reviewable.io:443/reviews/innovate-technologies/itframe/17#-KNY_lbE3wbfE7oEbqpQ:-KNYbNnNIlIW_PoPGuZW:-499146009) ([raw file](https://github.com/innovate-technologies/itframe/blob/e77aa355a068bf395e02de2fc538382502c8d9d8/http/internal/tunes.js#L144)):*
<details><summary><i>Previously, leoetlino (Léo Lam) wrote…</i></summary>
> there's no need for next for error handling in an async route handler. Just `throw` it
</details>
Done.

---

*[http/internal/tunes.js, line 148 \[r1\]](https://reviewable.io:443/reviews/innovate-technologies/itframe/17#-KNY_sI1y7HVXIAGPAaX:-KNYbOMhGiVI0-S8T4HB:-499146009) ([raw file](https://github.com/innovate-technologies/itframe/blob/e77aa355a068bf395e02de2fc538382502c8d9d8/http/internal/tunes.js#L148)):*
<details><summary><i>Previously, leoetlino (Léo Lam) wrote…</i></summary>
> space after { and before }
</details>
Done.

---

*[http/internal/tunes.js, line 151 \[r1\]](https://reviewable.io:443/reviews/innovate-technologies/itframe/17#-KNY_yydKCRX0DiatCsL:-KNYbP20fqxL9S5PPJqE:-499146009) ([raw file](https://github.com/innovate-technologies/itframe/blob/e77aa355a068bf395e02de2fc538382502c8d9d8/http/internal/tunes.js#L151)):*
<details><summary><i>Previously, leoetlino (Léo Lam) wrote…</i></summary>
> actually, you don't need to try and catch if you aren't going to change the error. just let the wrapper function catch it and pass it to the error handler.
> 
> Since this isn't the case here, you can get rid of the whole try/catch block.
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/innovate-technologies/itframe/17)*

**Comment by leoetlino** [comment]
Reviewed 2 of 2 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[http/internal/tunes.js, line 142 \[r3\]](https://reviewable.io:443/reviews/innovate-technologies/itframe/17#-KNYb_STDZ7cJ8_dWmJ2:-KNYb_STDZ7cJ8_dWmJ3:-1788241097) ([raw file](https://github.com/innovate-technologies/itframe/blob/57cda7d8d57b788d73dbfd9cbc07e088404eda3a/http/internal/tunes.js#L142)):*
> ```JavaScript
>     });
> 
>     app.get("/tunes/is-link-in-use/" + config.tunesKey, wrap(async (req, res, next) => {
> ```

next is an unused variable now.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/innovate-technologies/itframe/17)*

**Comment by meyskens** [comment]
Review status: 1 of 2 files reviewed at latest revision, 2 unresolved discussions.

---

*[http/internal/tunes.js, line 142 \[r3\]](https://reviewable.io:443/reviews/innovate-technologies/itframe/17#-KNYb_STDZ7cJ8_dWmJ2:-KNYcHJ1YMMJJ441cqXq:-499146009) ([raw file](https://github.com/innovate-technologies/itframe/blob/57cda7d8d57b788d73dbfd9cbc07e088404eda3a/http/internal/tunes.js#L142)):*
<details><summary><i>Previously, leoetlino (Léo Lam) wrote…</i></summary>
> next is an unused variable now.
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/innovate-technologies/itframe/17)*

**Comment by leoetlino** [comment]
:lgtm:

---

Reviewed 1 of 1 files at r4.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/innovate-technologies/itframe/17#-:-KNYcZSjCLP9h8p7EsDy:1417088721)*

**Pull request merged**